processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -160,6 +160,15 @@ public NestedTargetPropertyMappingHolder build() {
                         groupedByTP.singleTargetReferences.get( targetProperty )
                     );
 
+                    // If there are multi source parameters that are mapped to the target reference or
+                    // there are multiple source references to the target reference then we need to force an
+                    // update method. The reason for this is that multiple sources have effect on the target.
+                    // See Issue18282Test for more info.
+                    boolean forceUpdateMethod =
```

Comment:
are there really 2 booleans required to fix this?

Reply:
It is one boolean, but I wanted to extract one that forces an update method and the other one with the nested reference